mozilla_django_oidc_db/setupconfig/bootstrap.py

```diff
@@ -0,0 +1,80 @@
+from django.conf import settings
+from django.contrib.auth.models import Group
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.test import RequestFactory
+
+from django_setup_configuration.config_settings import ConfigSettings
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed, SelfTestFailed
+
+from ..forms import OIDCSetupConfigForm
+from ..models import OpenIDConnectConfig
+from ..views import OIDCAuthenticationRequestView
+
+
+class AdminOIDCConfigurationStep(BaseConfigurationStep):
+    """
+    Configure admin login via OpenID Connect
+    """
+
+    verbose_name = "Configuration for admin login via OpenID Connect"
+
+    config_settings = ConfigSettings(
+        enable_setting="OIDC_DB_CONFIG_ENABLE",
+        display_name="Admin OIDC Configuration",
+        namespace="OIDC_DB",
+        models=[OpenIDConnectConfig],
+        update_fields=True,
+        required_settings=["OIDC_DB_SETUP_CONFIG_ADMIN_AUTH"],
+    )
+
+    def is_configured(self) -> bool:
+        return OpenIDConnectConfig.get_solo().enabled
+
+    def configure(self):
+        config = OpenIDConnectConfig.get_solo()
+
+        all_settings = {
+            "sync_groups": config.sync_groups,
+            "oidc_use_nonce": config.oidc_use_nonce,
+        }
+
+        all_settings.update(settings.OIDC_DB_SETUP_CONFIG_ADMIN_AUTH)
+
+        if "default_groups" in all_settings:
+            groups = all_settings["default_groups"]
+            for group_name in groups:
+                Group.objects.get_or_create(name=group_name)
+            all_settings["default_groups"] = Group.objects.filter(name__in=groups)
+
+        all_settings["enabled"] = True
+
+        form = OIDCSetupConfigForm(
+            instance=config,
+            data=all_settings,
+        )
+        if not form.is_valid():
+            raise ConfigurationRunFailed(
+                f"Something went wrong while saving configuration: {form.errors.as_json()}"
+            )
+
+        form.save()
+
+    def test_configuration(self):
+
+        request_factory = RequestFactory()
+        request = request_factory.get("/")
+
+        middleware = SessionMiddleware(lambda x: None)
```

Comment:
I *think* type checkers don't like this because the callable doesn't produce a response.

Easiest way is shutting them up:

```suggestion
        middleware = SessionMiddleware(lambda x: None)  # type: ignore
```